apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.IntegrationTests/PartitionedReadTests.cs

```diff
@@ -26,101 +26,96 @@ namespace Google.Cloud.Spanner.Data.IntegrationTests
+                    partitions => PartitionedReadTests.DistributedReadWorkerAsync(partitions, transactionId)));
+            Assert.Equal(_fixture.TableRowCount, partitionNumRows.Sum());
+        }
 
+        private async Task<(IEnumerable<CommandPartition> partitions, TransactionId transactionId)> GetPartitionsAsync(Func<SpannerConnection, SpannerCommand> commandFactory, PartitionOptions options)
+        {
             using var connection = new SpannerConnection(_fixture.ConnectionString);
             await connection.OpenAsync();
 
             using var transaction = await connection.BeginReadOnlyTransactionAsync();
-            using var cmd = query
-                ? connection.CreateSelectCommand($"SELECT * FROM {_fixture.TableName}")
-                : connection.CreateReadCommand(_fixture.TableName, ReadOptions.FromColumns(_fixture.ColumnNames), KeySet.All);
             transaction.DisposeBehavior = DisposeBehavior.CloseResources;
-            cmd.Transaction = transaction;
-            var partitions = await cmd.GetReaderPartitionsAsync(1000);
-            var transactionId = transaction.TransactionId;
-
-            //we simulate a serialization/deserialization step in the call to the subtask.
-            await Task.WhenAll(partitions.Select(
-                    x => DistributedReadWorkerAsync(CommandPartition.FromBase64String(x.ToBase64String()),
-                                    TransactionId.FromBase64String(transactionId.ToBase64String()))))
-                .ConfigureAwait(false);
-
-            Assert.Equal(numRows, _rowsRead);
-        }
 
-        private async Task<int> GetNumberOfRows()
-        {
-            using var connection = _fixture.GetConnection();
-            using var cmd = connection.CreateSelectCommand($"SELECT COUNT(*) FROM {_fixture.TableName}");
-            return await cmd.ExecuteScalarAsync<int>();
+            using var cmd = commandFactory(connection);
+            cmd.Transaction = transaction;
+            return (await cmd.GetReaderPartitionsAsync(options), transaction.TransactionId);
         }
 
         /// <summary>
         /// TransactionId and CommandPartition are serializable.
         /// </summary>
-        private async Task DistributedReadWorkerAsync(CommandPartition readPartition, TransactionId id)
+        private static async Task<int> DistributedReadWorkerAsync(CommandPartition readPartition, TransactionId id)
         {
+            int readRows = 0;
+
             // Note: we only use state provided by the arguments here.
-            using (var connection = new SpannerConnection(id.ConnectionString))
+            using var connection = new SpannerConnection(id.ConnectionString);
+            using var transaction = connection.BeginReadOnlyTransaction(id);
+            using var cmd = connection.CreateCommandWithPartition(readPartition, transaction);
+            using (var reader = await cmd.ExecuteReaderAsync().ConfigureAwait(false))
```

Comment:
Use the block-less form of the using statement here, while we're modifying this code anyway?